.github/workflows/python-package-poetry.yml

```
@@ -0,0 +1,61 @@
+name: Python Package (Poetry)
+
+on:
+  push:
+    branches:
+      - main
+      - prod-deploy
+  pull_request:
+    paths:
+      - "backend/**"
+      - pyproject.toml
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9.16'
```

```suggestion
          python-version: '3.9.17'
```

should we use 3.9.17 to be consistent in our github actions @codingwithsurya